### PR TITLE
feat: Update `trestle md` to `trestle author` and introduce functionality for validating drawio metadata.

### PR DESCRIPTION
**Pull request by butler54**
Signed-off-by: Chris Butler <chris@thebutlers.me>

## Types of changes

- \[x\] Bug fix (non-breaking change which fixes an issue)
- \[x\] New feature (non-breaking change which adds functionality)
- \[x\] Breaking change (fix or feature that would cause existing functionality to change)
- \[x\] My code follows the code style of this project.
- \[x\] My change requires a change to the documentation.
- \[x\] I have updated the documentation accordingly.
- \[x\] I have added tests to cover my changes.
- \[x\] All new and existing tests passed.
- \[x\] All commits are signed-off.

This PR has a number of significant changes which have been introduced across the board.
- Enforce UTF-8 across all operating systems
- Change from trestle md to trestle author
- Improve unit tests in a number of places for trestle author functionality
- Introduce trestle author headers for validating headers only across an arbitrary set of file types (currently markdown and drawio).

Closes: #458 
Closes: #488 
Closes: #508

### CODE REVIEW

File: docs/trestle_author.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
+# trestle CLI for governance of authored documents
+
+## Overview
+
+The premise of trestle is to support managing compliance artifacts as code.
+When this is considered, many organisations using `{github|gitlab|bitbucket}` rely on markdown documents for documentary artifacts that may either directly or indirectly support compliance efforts.
+
+To support this trestle has the concept of 'governing' documents that are authored documents: Where structural conditions are enforced on the markdown documents to allow automation and to ensure business processes are met.
+
+Living in the `gitOps` world this capability is anchored with `markdown` files as the core of the workflows. Currently `drawio` files are also supported for a subset of enforcement mechanisms.
```
Comment (fsuits) [inline docs/trestle_author.md:10]: branding nit - I think it's GitOps

File: docs/trestle_author.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+my_other key with strange stuff?? my value
+```
+
+This capability, where available, is activated by `--governed-heading` or `-gh`
+
+### Drawio enforcement mechanisms
+
+Drawio or [diagrams.net](https://app.diagrams.net/) is a diagramming platform which has significant use for architecture diagrams. In the context of governance of content trestle is supporting enforcement of metadata.
+
+Drawio (or `mxgraph`) files have a set of data fields. In a drawio file this is available the edit menu as *edit data*. The diagram below shows how to access the (meta)data.
```
Comment (fsuits) [inline docs/trestle_author.md:122]: And I think draw.io needs the dot in there.

File: tests/trestle/core/commands/author/command_test.py [hunk 3]
```diff
@@ -25,60 +25,46 @@
...
 def test_failure_not_trestle(command_string, tmp_path: pathlib.Path) -> None:
     """Test for failure based on not in trestle directory."""
     with mock.patch.object(sys, 'argv', command_string.split()):
         with pytest.raises(SystemExit) as wrapped_error:
             trestle.cli.run()
             # FIXME: Needs to be changed once implemented.
-            assert wrapped_error == SystemExit
-            assert wrapped_error.code == 1
+        assert wrapped_error.type == SystemExit
+        assert wrapped_error.value.code == 1
```
Comment (fsuits) [inline tests/trestle/core/commands/author/command_test.py:70]: Wow - it was good to catch this.  A lot of them...

File: scripts/experiments/tanium_ben.py [hunk 4]
```diff
@@ -51,7 +51,7 @@ def read_file(in_file: str):
     file_path = Path(in_file)
     data = []
     count = 0
-    with open(file_path) as fp:
+    with open(file_path, encoding='utf8') as fp:
```
Comment (fsuits) [inline scripts/experiments/tanium_ben.py:54]: It doesn't really matter but all these utf8's could be const.FILE_ENCODING - but maybe you want to keep scripts separate from trestle modules.

**Review by fsuits — APPROVED** [review]
LGTM.  Some minor suggestions - please consider before merge.

File: docs/trestle_author.md (hunk 1 above)
Comment (butler54) [reply docs/trestle_author.md:10]: Fixed.

File: docs/trestle_author.md (hunk 2 above)
Comment (butler54) [reply docs/trestle_author.md:122]: I'm happy without the dot as the file extension is `.drawio`.

File: tests/trestle/core/commands/author/command_test.py (hunk 3 above)
Comment (butler54) [reply tests/trestle/core/commands/author/command_test.py:70]: Yes it did cause some problems.. a bit of a nasty find.

File: scripts/experiments/tanium_ben.py (hunk 4 above)
Comment (butler54) [reply scripts/experiments/tanium_ben.py:54]: Fixed where the code is importing trestle. If not I have used utf-8 as we might end up with circular dependencies.

**Pull request merged**